presto-main/src/main/java/com/facebook/presto/operator/aggregation/SumAggregation.java

```diff
@@ -0,0 +1,153 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.byteCode.DynamicClassLoader;
+import com.facebook.presto.metadata.FunctionInfo;
+import com.facebook.presto.metadata.FunctionRegistry;
+import com.facebook.presto.metadata.ParametricAggregation;
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.operator.aggregation.state.AccumulatorState;
+import com.facebook.presto.operator.aggregation.state.AccumulatorStateFactory;
+import com.facebook.presto.operator.aggregation.state.AccumulatorStateSerializer;
+import com.facebook.presto.operator.aggregation.state.NullableDoubleState;
+import com.facebook.presto.operator.aggregation.state.NullableDoubleStateSerializer;
+import com.facebook.presto.operator.aggregation.state.NullableLongState;
+import com.facebook.presto.operator.aggregation.state.NullableLongStateSerializer;
+import com.facebook.presto.operator.aggregation.state.StateCompiler;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.StandardErrorCode;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.google.common.collect.ImmutableList;
+
+import java.lang.invoke.MethodHandle;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.metadata.Signature.typeParameter;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.INPUT_CHANNEL;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.STATE;
+import static com.facebook.presto.operator.aggregation.AggregationUtils.generateAggregationName;
+import static com.facebook.presto.util.Reflection.methodHandle;
+
+public class SumAggregation
+        extends ParametricAggregation
+{
+    private static final MethodHandle LONG_INPUT_FUNCTION = methodHandle(SumAggregation.class, "input", NullableLongState.class, long.class);
+    private static final MethodHandle DOUBLE_INPUT_FUNCTION = methodHandle(SumAggregation.class, "input", NullableDoubleState.class, double.class);
+
+    public static final SumAggregation SUM_AGGREGATION = new SumAggregation();
+    private static final String NAME = "sum";
+    private static final Signature SIGNATURE = new Signature(NAME, ImmutableList.of(typeParameter("E")), "E", ImmutableList.of("E"), false, false);
```

Comment:
This doesn't work. E will bind to any type, so you could call `SUM(varchar_column)` which will then fail